Minutes from Nov 8, 2021 (#199)
solid/authentication-panel: meetings/2021-11-08.md

* AC: Do you have an opinion about scopes, in particular in the WWW-Authenticate header?
* eP: do you mean whether the RS should communicate the scope parameter?
* AC: Yes
* AC: Agreed, such scopes may belong more to the AuthZ spec than to the AuthN spec. However, there is a lot of overlap between specs there. Should Solid-OIDC mention this at least?
* eP: we should only need to say that WWW-Authenticate is the mechanism to discover how an entity can get access to a resource.
* AC: This would make the specification stronger, and simpler, with the caveat that we should be careful about how to redirect implementers to additional AuthZ specifications without being too overwhelming to discover how to approach these scopes.
* eP: one issue is we then depend on the authorization spec to define that. The ecosystem would need the authorization spec as a MUST.
* AC: What would be the next specs for the Solid-OIDC spec ?
* eP: next steps would be to introduce the authorization server. E.g., www-authenticate `as_uri`
* ...: remove access token
* ...: bind DPoP token to ID token
* ...: describe claims pushing, `id_token` used as claim

> * AC: We'll want to be communicate clearly to the wider Solid community what these changes entail, and what path should be followed for transition.

Reviewer: ```suggestion
* AC: We'll want to communicate clearly to the wider Solid community what these changes entail, and what path should be followed for transition.
```

* ...: We will want to make sure there is client support before asking servers to change
* eP: as a maintainer of a client library, how do you see this transition period? Do you need clear documentation of these changes?
* NAS: it is a bit early to asses the impact. It isn't too problematic to have the two live side-by-side. An RS can advertise this capability. The transition period is more for the client libraries to adapt
* eP: if both the client and server have a transition period, then there is a path for both to move forward
* ...: there was discussion from a few weeks ago about keeping access tokens in a closure, so it might also be possible to store the ID token like that
* NAS: there are some things to double-check there. For refresh, we rely on the storage of an ID token stored in local storage.
* eP: there is an open PR that starts this (??? add URL). What would be the best workflow
* AC: one first step might be to describe the DPoP-bound ID token
* eP: with access tokens, we have a notion of refreshing. how will that work with ID tokens?
* NAS: what would be the expected lifetime of an ID token? Do we expect that the lifespan would be longer lived?

> * AC: It should be short-lived too, because it you can trade the ID token for an access token, then there is no difference between having a long-lived AT and a long-lived ID Token

Reviewer: ```suggestion
* AC: It should be short-lived too, because if you can trade the ID token for an access token, then there is no difference between having a long-lived AT and a long-lived ID Token
```

* eP: we don't currently have a way to revoke ID tokens. In practice, is this hours or minutes?
* NAS: ten minutes? also for ID tokens
* AC: ESS expires tokens after 30min, CSS after 1 hour, NSS after 2 weeks
* Barath: is there need to have standard time?

> * AC: There is no need for a standard lifespan as long as there is a shared convention to refresh them. The industry convension is that they are short-lived (an hour or less). The more sensitive the context is, the shorter the lifespan

Reviewer: ```suggestion
* AC: There is no need for a standard lifespan as long as there is a shared convention to refresh them. The industry convention is that they are short-lived (an hour or less). The more sensitive the context is, the shorter the lifespan
```

* BR: could we customize this in real-time? You could imagine that in offline/partitioned operation, you might want to change that operation time. For instance, if you know in advance that you can't refresh
* AC: An approach would be to build up state locally and then flush that state to the RS once re-authenticated.
* eP: let's continue to flesh this out over the week and move to the next topic.

ACTION: eP: create issue about refreshing an expired ID token

### Test Suite

https://github.com/solid/solid-oidc/issues/22

> * AC: Some progress has been made. A browser-based static webapp is available, where you specify the target Solid-OIDC issuer, and which runs a set of tests covering parts of the OAuth2.0, OpenID and Solid-OIDC specs.

Reviewer: ```suggestion
* AC: Some progress has been made. A browser-based static webapp is available, where you specify the target Solid-OIDC issuer, and the webapp runs a set of tests covering parts of the OAuth2.0, OpenID, and Solid-OIDC specs.
```

* ...: No distinction between which spec is tested by which test.
* ...: No distinction between MAY and MUST

> * ...: The test report is generated incrementally, and the question of mainting state between the requesuts is an issue.

Reviewer: ```suggestion
* ...: The test report is generated incrementally, and the question of maintaining state between the requests is an issue.
```

* ...: Approaches with iframes and popups (still working out the details) ...
* ...: The individual tests are not as much of an issue as maintaining the authenticated session.
* PE: What is the state of automation
* AC: For automation, `client_credentials` are great, but they are not required by the spec. This creates a tension between the spec, which only requires
* ...: `authorization_code` flow, and tests that would require the `client_credentials`, which is why brower-based tests are good.
* PE: What kinds of interaction is required by a user?
* AC: The manual steps are authentication, and specifying the OIDC provider under test. How the user does authenticate to the OP is out of scope of the spec, so no specific assumptions can be made there.
* AC: What we may want to do is
* ...: For the tests harness, a specific authentication flow is expected. The browser-based tests could be rewritten easily to fit into these harnesses.
* ...: The things that are mostly tested are:
* ...: - Is discovery data accurate
* ...: - What is present in the access token

> * ...: - The surrounding context (e.g. PKCE, DPoP) is well supported

Reviewer: ```suggestion
* ...: - The surrounding context (e.g., PKCE, DPoP) is well supported
```
